File: pmid_test.ts
//
// Test our PMID methods
//
import { assertEquals, assertNotEquals } from "@std/assert";

import { normalizePMID, validatePMID } from "./pmid.ts";
import { verifyPMID } from "./pmid_record.ts";

Deno.test("test pmid", async () => {
  const varified_ids: string[] = [
    "34125777",
    "PMID: 25591183",
  ];

  for (const id of varified_ids) {
    // Normalize
    const normalized = normalizePMID(id);
    assertNotEquals(normalized, undefined);
    console.log(`Normalized PMID: ${normalized}`);
    // Validate
    assertEquals(validatePMID(id), true);
    // Verify is via pmid.org
    const isOK: boolean = await verifyPMID(id);
    assertEquals(isOK, true);
  }
});

File: pmid_test.go
package metadatatools

import "testing"

func TestPMID(t *testing.T) {
	if testing.Short() {
		t.Skip("skipping network test in short mode")
	}

	verifiedIDs := []string{
		"34125777",
		"PMID: 25591183",
	}

	for _, id := range verifiedIDs {
		normalized := NormalizePMID(id)
		if normalized == "" {
			t.Errorf("NormalizePMID(%q) returned an empty string", id)
		}
		t.Logf("Normalized PMID: %s", normalized)

		if !ValidatePMID(id) {
			t.Errorf("ValidatePMID(%q) = false, expected true", id)
		}

		if !VerifyPMID(id) {
			t.Errorf("VerifyPMID(%q) = false, expected true", id)
		}
	}
}
